test/unit/org/apache/cassandra/utils/memory/MemoryUtilTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils.memory;
+
+import java.lang.management.BufferPoolMXBean;
+import java.lang.management.ManagementFactory;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MemoryUtilTest
+{
+
+    @Test
+    public void testClean()
+    {
+        // We assert direct pool state changes as no ByteBuffer state changes are observable as a result of cleaner
+        // executing.
+        BufferPoolMXBean directPool = getDirectBufferPool();
+
+        int bufferSize = 1024 * 1024; // some non-insignificant size
+        ByteBuffer original = ByteBuffer.allocateDirect(bufferSize);
+
+        long memoryUsedBefore = directPool.getMemoryUsed();
+        MemoryUtil.clean(original);
+        long memoryUsedAfter = directPool.getMemoryUsed();
+
+        Assert.assertEquals("Direct memory used should decrease by buffer capacity",
+                            // Allow 5% tolerance for other activities
+                            memoryUsedBefore - bufferSize, memoryUsedAfter, bufferSize * 0.05);
+    }
+
+    @Test
+    public void testCleanViewDoesNotThrow()
+    {
+        ByteBuffer original = ByteBuffer.allocateDirect(16);
+
+        ByteBuffer slice = original.slice();
+        MemoryUtil.clean(slice);
+
+        ByteBuffer duplicate = original.duplicate();
+        MemoryUtil.clean(duplicate);
```

Comment:
The test should also check that the original is still readable/writable. In practice the memory will probably still be readable/writable even when deallocated, but oh well.

Reply:
Actually if you made these buffers large enough they would get mmaped and unmapped explicitly in at least some allocators and then you could actually check reliably if they are supposed to still be readable/writable.

I don't know what the common default threshold, but if you do say 64mb I am pretty sure if an allocator falls back to mmap it will do it.

For glibc
```
              The lower limit for this parameter is 0.  The upper limit
              is DEFAULT_MMAP_THRESHOLD_MAX: 512*1024 on 32-bit systems
              or 4*1024*1024*sizeof(long) on 64-bit systems.
```

I'll ask for this change since it's trivial to change the allocation size in the test.

Reply:
This is a great idea.